docs/guide/topics/permissions.rst

```diff
@@ -0,0 +1,219 @@
+
+.. code-block:: python3
+
+    async def create_new_role(name: str, mentionable: bool, guild: discord.Guild) -> discord.Role:
+        await guild.create_role(name=name, mentionable=mentionable, reason="New game!")
+
+This rather short example just creates a new role within the :class:`~Guild` passed to this method.
+
+Let's flesh it out more, and use the :meth:`~Guild.create_role` method to it's full capability with a set of :class:`Permissions`.
+
+.. code-block:: python3
+
+    # So we want a new role, but let's give it permissions to do some things.
+    # We will define a set of `Permissions` that do what we need:
+    permissions = discord.Permissions()
+    permissions.send_messages = True
+    permissions.add_reactions = True
+    permissions.create_public_threads = True
+
+    # Above is now a working Permissions instance with the name `permissions`, we could also define it like so:
+    permissions = discord.Permissions(send_messages=True, add_reactions=True, create_public_threads=True)
+
+    # and we pass this to the role creation method!
+    await guild.create_role(name="My cool game", mentionable=True, permissions=permissions, reason="New game!")
+
+Now we have a brand new role, named ``"My cool game"`` and it now have the permissions we outlined above.
+
+Setting permissions on an existing channel
+-------------------------------------------
+
+Let's say you wanted to edit the permissions of an existing channel, perhaps to give a role more access.
+An example here is that my new moderator role needs the Manage Messages permission in the #general channel.
+We will be using the :meth:`~TextChannel.set_permissions` method:
+
+.. code-block:: python3
+
+    # Let's create a PermissionOverwrite instance that has some explicit permissions set:
+    overwrites = discord.PermissionOverwrite()
+    overwrites.update(send_messages=True, send_messages_in_threads=True)
+
+    # and now we update that with `manage_messages`:
+    overwrites.manage_messages = True
+
+    # we can now set those permissions on the channel:
+    await channel.set_permissions(moderator_role, overwrites=overwrites)
+
+This short example briefly goes over PermissionOverwrite objects, let's look more into those now:
+
+Using ``PermissionOverwrite`` objects during channel creation
+--------------------------------------------------------------
+
+We can also create channels with permissions set from the start.
+Let's create a text channel where only the moderator role, and someone trying to make a report can see the channel.
+
+We need to construct a mapping of ``Role | Member: PermissionOverwrite``:
+
+.. code-block:: python3
+
+    # We can now create the overwrites mapping necessary for this, you can do it in one dictionary instantiation, or spead it across multiple lines.
+
+    # I have added type annotations here so that the reader can understand what is happeining, and so type-checkers do too:
```

Comment:
```suggestion
    # I have added type annotations here so that both the reader and type-checkers can understand what is happening:
```
Also, this might conversely have the inverse effect of being too complicated for new readers. Might be a good idea to reconsider heavily typing examples.